src/meta/src/controller/catalog/drop_op.rs

```diff
@@ -138,6 +139,28 @@ impl CatalogController {
             }
         }
 
+        // 1. Detect the case where the iceberg table created from a shared source and drop source cascade.
+        // Currently, iceberg engine table doesn't support atomic ddl. To drop the source, we need to drop the iceberg table first.
+        //
+        // 2. Drop database with iceberg tables in it is not supported.
+        if object_type == ObjectType::Source || drop_database {
```

Comment:
Can we just check `if object_type != ObjectType::Table && drop_mode == DropMode::Cascade` here?

Reply:
I think only the source and database can have dependencies on tables.

Reply:
But what we actually care is only "the iceberg table is dropped as a dependency", we don't care who depends on the table. So I think checking `object_type != ObjectType::Table` is slightly more robust

Reply:
> But what we actually care is only "the iceberg table is dropped as a dependency", we don't care who depends on the table. So I think checking object_type != ObjectType::Table is slightly more robust

yep, it makes sense. Let me change it to `object_type != ObjectType::Table`